### PR TITLE
Return `mean(delta_t values in bin)` as `dts` for Structure Function

### PR DESCRIPTION
**Pull request by drewoldag**
In the main branch after we have established the bins for the time differences, we return the bin centers to the user as the parameter `dts`. However, in practice this value can be misleading, and it is more useful to return the mean of the values in a given bin as the parameter `dts`. 

This PR represents the work to gather those statistics for the bins. There were several unit tests that needed to be updated as well, but because we were using small, contrived examples, it was not difficult to verify that the new `dts` values were correct.

### CODE REVIEW

File: src/lsstseries/analysis/structure_function/base_calculator.py
Comment (smcguire-cmu) [inline src/lsstseries/analysis/structure_function/base_calculator.py:150]: Should the docstring for this function be updated to specify that it returns the `bin_means` now? I see it still mentions the edges (but that might be for another reason)

**Review by smcguire-cmu — COMMENTED** [review]
It looks like a simple enough change, and I don't think there's anything wrong with the code written here, but I don't think I really know enough about how it fits in with the rest of the codebase to properly review it.

**Review by smcguire-cmu — COMMENTED** [review]
It looks like a simple enough change, and I don't think there's anything wrong with the code written here, but I don't think I really know enough about how it fits in with the rest of the codebase to properly review it.

File: src/lsstseries/analysis/structure_function/base_calculator.py
Comment (drewoldag) [reply src/lsstseries/analysis/structure_function/base_calculator.py:150]: Good call out, I'll update the docstring.

Comment (drewoldag) [reply src/lsstseries/analysis/structure_function/base_calculator.py:150]: Updated

**Review by smcguire-cmu — APPROVED** [review]
Looks good to me, but might want to wait until approval from Doug or Neven. (Unless you've already talked about it)

**Review by nevencaplar — APPROVED** [review]
Looks good to me

**Pull request merged**
